pkg/client/client.go

```diff
@@ -0,0 +1,96 @@
+package client
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/solo-io/go-utils/contextutils"
+	api "github.com/solo-io/reporting-client/pkg/api/v1"
+	"google.golang.org/grpc"
+)
+
+const (
+	// set this env var to the string "true" to prevent usage from being reported
+	DisableUsageVar = "USAGE_REPORTING_DISABLE"
+)
+
+// a type that knows how to load the usage payload you want to report
+type UsagePayloadReader interface {
+	GetPayload() (map[string]string, error)
+}
+
+//go:generate mockgen -destination mocks/mock_usage_payload_reader.go -package mocks github.com/solo-io/reporting-client/pkg/client UsagePayloadReader
+//go:generate mockgen -destination mocks/mock_reporting_service_client.go -package mocks github.com/solo-io/reporting-client/pkg/api/v1 ReportingServiceClient
+
+type Client interface {
+	StartReportingUsage(ctx context.Context, interval time.Duration) context.CancelFunc
```

Comment:
Rather than returning a `context.CancelFunc` better to assume that the parent will take carte of this